src/features/walletconnect/components/WcSessionMananger/index.tsx

```diff
@@ -62,6 +66,21 @@ const WcSessionManager = ({ sessions, uri }: WcSessionManagerProps) => {
     setProposal(undefined)
   }, [proposal, walletConnect, setError])
 
+  // Try to clean-up when the user has open proposal,
+  // but for whatever reason refreshes the browser without rejecting/accepting it
+  // if we don't do this we often get "No matching key: session"
+  useEffect(() => {
+    const handleBeforeUnload = async () => {
+      if (proposal && isResponded.current === false) {
+        await onReject()
+      }
+    }
+
+    window.addEventListener('beforeunload', handleBeforeUnload)
```

Comment:
AFAIR beforeunload is supposed to return a boolean to prevent the browser from closing? Maybe we should use onunload?

Reply:
Thanks!